### ISSUE TITLE
Error when trying to read the `conanfile.txt`

### ISSUE THREAD

**Issue by knoxfighter** [conan-io/conan#4524]
In our Jenkins-system, jenkins created a projectdirectory, that contains an "@". After we generated the conanfile.txt, we cannot run conan, cause it doesn't find the "@ folder=True" at the end of the line.

The conanfile.txt parser, splits at the first "@" and when we habe an "@" in our directory name, the parsing fails.

Example:
/home/jenkins/workspace/conan_test@2/conanfile.txt
```
[generators]
cmake

[requires]
OpenSSL/1.1.1@conan/stable

[options]
OpenSSL:shared=True
OpenSSL:no_zlib=True

[imports]
"., LICENSE -> /home/jenkins/workspace/conan_test@2/install/licenses @ folder=True"
lib, * -> /home/jenkins/workspace/conan_test@2/g/install/lib
bin, * -> /home/jenkins/workspace/conan_test@2/install/bin
include, * -> /home/jenkins/workspace/conan_test@2/install/include
```

Run conan install:
`conan install . -g cmake -s build_type=Debug -s os=Linux -s arch=x86_64 -s compiler=gcc -s compiler.version=7 -s compiler.libcxx=libstdc++11 --build=missing`

Error:
ERROR: Invalid imports line: "., LICENSE -> /home/jenkins/workspace/conan_test@2/install/licenses @ folder=True"
EX: OpenCV/lib, * -> ./lib
Invalid imports. Unknown argument 2install/licenses @ folder

Code with error:
/conans/client/loader_txt.py:
https://github.com/conan-io/conan/blob/ab15630bf0675eb2b787b4aaad8f79a53b3580f6/conans/client/loader_txt.py#L75

Possible Solution:
Split it at the last "@" if possible or ignore things, that are inside `"` or `'`.

**Comment by lasote** [comment]
Thanks for reporting.
It should be an `rsplit` and matching the right part with a regular expression of: `param=value` values separated or not with spaces.
How bad is this blocking to you?

**Comment by lasote** [comment]
It will be released at 1.12.3
